omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

```diff
@@ -357,27 +431,37 @@ def draw_polygon(self, shape, closed=True):
         if closed:
             for point in points[0:2]:
                 p.lineTo(point[0], point[1])
-        self.canvas.drawPath(p, fill=0, stroke=1)
+        self.canvas.drawPath(p, fill=fill, stroke=1)
+
+        self.draw_shape_label(shape, Bounds(*points))
 
     def draw_polyline(self, shape):
         self.draw_polygon(shape, False)
 
     def draw_ellipse(self, shape):
         stroke_width = shape['strokeWidth']
         c = self.panel_to_page_coords(shape['x'], shape['y'])
+
```

Comment:
that means that we wont' be able to handle shapes covering partially the image

Reply:
No, that's always been the case with PDF export. Shapes that extend outside of the cropped image panel are drawn on the page without being confined to the panel that they come from. Haven't found a way to be able to work around that.

Reply:
Thanks for the clarification @will-moore